src/Moq/Interception/InterceptionAspects.cs

```diff
@@ -150,15 +150,22 @@ public static bool Handle(Invocation invocation, Mock mock)
 
 						if (mock.CallBase && !invocation.Method.IsAbstract)
 						{
-							invocation.ReturnBase();
-							return true;
+							if(!mock.Setups.HasEventSetup)
```

Comment:
Reading `mock.Setups.HasEventSetup` each time could lead to the problem when event setup has been added just in between `invocation.Return()` and `return` statement. It could lead to the problems later during the pipeline when invocation should be returned(either matched or non matched). Tmp variable should be used bool `hasEventSetup = mock.Setups.HasEventSetup`...